src/test/java/core/basesyntax/dao/RegistrationServiceTest.java

```diff
@@ -0,0 +1,118 @@
+package core.basesyntax.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationValidationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void register_ValidUser_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("password123");
+        user.setAge(25);
+
+        User registeredUser = registrationService.register(user);
+
+        assertNotNull(registeredUser);
+        assertEquals("testlogin", registeredUser.getLogin());
+        assertEquals("password123", registeredUser.getPassword());
+        assertEquals(25, registeredUser.getAge());
+        assertNotNull(registeredUser.getId());
+    }
+
+    @Test
+    void validLogin_Not_Ok() {
+        User user = new User();
+        user.setLogin("log");
+        user.setPassword("password123");
+        user.setAge(25);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user);
+        }, "RegistrationValidationException "
+                + "should be thrown if the login has less than 6 characters");
+    }
+
+    @Test
+    void validPassword_Not_Ok() {
+        User user = new User();
+        user.setLogin("testlogin");
+        user.setPassword("pass");
+        user.setAge(25);
+
+        assertThrows(RegistrationValidationException.class, () -> {
+            registrationService.register(user);
+        }, "RegistrationValidationException "
+                + "should be thrown if the password has less than 6 characters");
+    }
+
+    @Test
+    void validAge_Not_Ok() {

```

Comment:
The method name `validAge_Not_Ok` does not conform to the naming convention. It should be named something like `register_ageLessThanRequired_notOk` to accurately represent the test case.